Mathlib/Tactic/Linter/TextBased.lean

```diff
@@ -74,6 +84,29 @@ def StyleError.errorMessage (err : StyleError) : String := match err with
     endings (\n) instead"
   | trailingWhitespace => "This line ends with some whitespace: please remove this"
   | semicolon => "This line contains a space before a semicolon"
+  | StyleError.unicodeVariant s selector pos =>
+    let variant := if selector == UnicodeVariant.emoji then
+      "emoji"
+    else if selector == UnicodeVariant.text then
+      "text"
+    else
+      "default"
+    let oldHex := " ".intercalate <| s.data.map printCodepointHex
+    match s, selector with
+    | ⟨c₀ :: []⟩, some sel =>
+      let newC : String := ⟨[c₀, sel]⟩
+      let newHex := " ".intercalate <| newC.data.map printCodepointHex
+      s!"missing unicode variant-selector at char {pos}: \"{s}\" ({oldHex}). \
```

Comment:
nit: isn't English grammar to write "variant selector" instead? Can you audit the whole PR for this, please?

Reply:
done: 7f27e98b3dd558fdfd70cf7e798f746d9335b677